extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/InstrumenterVertxTracer.java

```diff
@@ -0,0 +1,164 @@
+package io.quarkus.opentelemetry.runtime.tracing.vertx;
+
+import static io.quarkus.opentelemetry.runtime.QuarkusContextStorage.ACTIVE_CONTEXT;
+
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.quarkus.opentelemetry.runtime.QuarkusContextStorage;
+import io.quarkus.opentelemetry.runtime.tracing.vertx.OpenTelemetryVertxTracer.SpanOperation;
+import io.vertx.core.Context;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.impl.headers.HeadersAdaptor;
+import io.vertx.core.http.impl.headers.HeadersMultiMap;
+import io.vertx.core.spi.tracing.SpanKind;
+import io.vertx.core.spi.tracing.TagExtractor;
+import io.vertx.core.spi.tracing.VertxTracer;
+import io.vertx.core.tracing.TracingPolicy;
+
+@SuppressWarnings("unchecked")
+interface InstrumenterVertxTracer<REQ, RESP> extends VertxTracer<SpanOperation, SpanOperation> {
+    @Override
+    default <R> SpanOperation receiveRequest(
+            final Context context,
+            final SpanKind kind,
+            final TracingPolicy policy,
+            final R request,
+            final String operation,
+            final Iterable<Map.Entry<String, String>> headers,
+            final TagExtractor<R> tagExtractor) {
+
+        Instrumenter<REQ, RESP> instrumenter = getReceiveRequestInstrumenter();
+        io.opentelemetry.context.Context parentContext = context.getLocal(ACTIVE_CONTEXT);
```

Comment:
This will fail when the checks are included if you are not on a duplicated context.